### PR TITLE
Add the possibility to use the Environment from the `ManagedLookup` in the `ManagedCache`

### PR DESCRIPTION
**Pull request by guizmaii**
To build a `ManagedCache`, you need to provide a `ManagedLookup`.
A `ManagedLookup` is defined as is:
```scala
final case class ManagedLookup[-Key, -Environment, +Error, +Value](lookup: Key => ZManaged[Environment, Error, Value])
    extends (Key => ZManaged[Environment, Error, Value])
```
But the `ManagedCache::get` method was returning a `Managed[Error, Value]` which was preventing us to use the `Environment` from the `ManagedLookup`

Now, the `ManagedCache::get` is returning `ZManaged[Environment, Error, Value]`

WDYT?

### CODE REVIEW

File: zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala [hunk 1]
```diff
@@ -69,43 +69,43 @@ object ManagedCache {
...
+  )(timeToLive: Exit[Error, Value] => Duration): UManaged[ManagedCache[Key, Env, Error, Value]] =
     ZManaged.make(buildWith(capacity, managedLookup, clock)(timeToLive))(_.invalidateAll)
 
-  private def buildWith[Key, Environment, Error, Value](
+  private def buildWith[Key, Env, Error, Value](
     capacity: Int,
-    managedLookup: ManagedLookup[Key, Environment, Error, Value],
+    managedLookup: ManagedLookup[Key, Env, Error, Value],
     clock: Clock
-  )(timeToLive: Exit[Error, Value] => Duration): URIO[Environment, ManagedCache[Key, Error, Value]] =
```
Comment (guizmaii) [inline zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala:102]: I'm not sure why we were returning a `URIO[Environment, ...]` 🤔

Do we want to keep this `Environment`? If yes, we'll need two envs for this function.

File: zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala [hunk 2]
```diff
@@ -171,45 +171,46 @@ object ManagedCache {
...
+                      }
+                  }
                 }
               }
             }
           }
-        }
 
-        override def refresh(k: Key): IO[Error, Unit] = lookupValueOf(k).memoize.flatMap { managed =>
+        override def refresh(k: Key): ZIO[Env, Error, Unit] = lookupValueOf(k).memoize.flatMap { managed =>
```
Comment (guizmaii) [inline zio-cache/shared/src/main/scala/zio/cache/ManagedCache.scala:213]: I'd have prefered not to have to change this but I need to apparently

**Comment by adamgfraser** [comment]
@guizmaii Can you say a little more about what you are trying to do? The original design here was to access the environment once when the cache was created to avoid having to access it on every call. Have an idea on how we can support what I think you are trying to do with a more minimal change.

**Comment by guizmaii** [comment]
@adamgfraser 

> Can you say a little more about what you are trying to do?

Well, we have something very special, but it doesn't seem solvable differently (AFAIK)
We don't need/don't have this `Env` value to get an instance of the cached resource, but we need this `Env` to use it

**Comment by guizmaii** [comment]
@adamgfraser That doesn't seem to work for our case, sadly. But I've found another trick.
It's related to this previous conversation: https://discord.com/channels/629491597070827530/787129044336508988/998923648666705982

The trick is to have a `LocalManagedCache` "tuned" interface and to use it:

```
abstract class LocalManagedCache[-K, +E, +A] {

  def cacheStats: UIO[CacheStats]

  def contains(key: K): UIO[Boolean]

  def entryStats(key: K): UIO[Option[EntryStats]]

  def get(key: K, myAdditional: Param): Managed[E, A]

  def refresh(key: K): IO[E, Unit]

  def invalidate(key: K): UIO[Unit]

  def invalidateAll: UIO[Unit]

  def size: UIO[Int]
}

ManagedCache
  .makeWith(
    capacity = ...,
    managedLookup = ...
  )(timeToLive = ...)
  .map(cache => 
    new LocalManagedCache {
      override def cacheStats: UIO[CacheStats] = cache.cacheStats

      override def contains(key: Key): UIO[Boolean] = cache.contains(key)

      override def entryStats(key: Key): UIO[Option[EntryStats]] = cache.entryStats(key)

      override def get(key: Key, myAdditional: Param): Managed[Throwable, MyResource] =
            cache.get(key).map(myResource => new InvalidatorMyResource(cache, myResource, myAdditional))

      override def refresh(key: Key): IO[Throwable, Unit] = cache.refresh(key)

      override def invalidate(key: Key): UIO[Unit] = cache.invalidate(key)

      override def invalidateAll: UIO[Unit] = cache.invalidateAll

      override def size: UIO[Int] = cache.size
    }
  )
```

**Comment by guizmaii** [comment]
@adamgfraser Thanks for the help! ❤️

**Pull request closed without merging**